runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesPlugin.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2017, Devin French <https://github.com/devinfrench>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.teamcapes;
+
+import com.google.inject.Binder;
+import com.google.inject.Provides;
+import net.runelite.api.Client;
+import net.runelite.api.GameState;
+import net.runelite.api.Player;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.task.Schedule;
+import net.runelite.client.ui.overlay.Overlay;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@PluginDescriptor(
+	name = "Team capes plugin"
+)
+public class TeamCapesPlugin extends Plugin
+{
+	@Inject
+	@Nullable
+	Client client;
+
+	@Inject
+	TeamCapesConfig config;
+
+	@Inject
+	TeamCapesOverlay teamCapesOverlay;
+
+	private Map<Integer, Integer> teams = new HashMap<>();
+
+	@Override
+	public void configure(Binder binder)
+	{
+		binder.bind(TeamCapesOverlay.class);
+	}
+
+	@Provides
+	TeamCapesConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(TeamCapesConfig.class);
+	}
+
+	@Override
+	public Overlay getOverlay()
+	{
+		return teamCapesOverlay;
+	}
+
+	@Schedule(
+		period = 1800,
+		unit = ChronoUnit.MILLIS
+	)
+	public void update()
+	{
+		teams.clear();
```

Comment:
Maybe move this part under the check? I know that clearing is not expensive on empty list, but it is still overhead without reason when the plugin is disabled.